src/main/java/io/reactivex/internal/operators/flowable/FlowableGenerate.java

```diff
@@ -122,7 +125,14 @@ public void request(long n) {
                         dispose(s);
                         return;
                     }
-
+                    //Multiple production threads cannot asynchronously
+                    //invoke synchronous delivery element methods at the same time
+                    if (!hasNext) {
```

Comment:
Not sure what you want here as the multi-call to `onNext` is already covered: https://github.com/ReactiveX/RxJava/blob/a9fe6333099f47babbab1c622dca2685f9aaf011/src/main/java/io/reactivex/internal/operators/flowable/FlowableGenerate.java#L177